honeybee_radiance_command/dctimestep.py

```diff
@@ -0,0 +1,323 @@
+        """Return a class instance explicitly for daylight coefficient calcs"""
+        dc_cls = cls(options=options, output=output, sky_vector=sky_vector,
+                     day_coef_matrix=day_coef_matrix)
+        dc_cls.is_daylight_coef_calc = True
+        return dc_cls
+
+    @classmethod
+    def for_direct_sun_calc(cls, options=None, output=None,
+                            sun_vector=None, sun_coef_matrix=None):
+        """Return a class instance explicitly for sun coefficient calcs"""
+        sc_cls = cls(options=options, output=output, sun_vector=sun_vector,
+                     sun_coef_matrix=sun_coef_matrix)
+        sc_cls.is_direct_sun_calc = True
+        return sc_cls
+
+    @classmethod
+    def for_three_phase_calc(cls, options=None, output=None,
+                             sky_vector=None, view_matrix=None, t_matrix=None,
+                             daylight_matrix=None):
+        """Return a class instance explicitly for three phase calcs"""
+        tph_cls = cls(options=options, output=output, sky_vector=sky_vector,
+                      view_matrix=view_matrix, t_matrix=t_matrix,
+                      daylight_matrix=daylight_matrix)
+        tph_cls.is_three_phase_calc = True
+        return tph_cls
+
+    @classmethod
+    def for_four_phase_calc(cls, options=None, output=None,
+                            sky_vector=None, view_matrix=None, t_matrix=None,
+                            facade_matrix=None, daylight_matrix=None):
+        """Return a class instance explicitly for three phase calcs"""
+        fph_cls = cls(options=options, output=output, sky_vector=sky_vector,
+                      view_matrix=view_matrix, t_matrix=t_matrix,
+                      daylight_matrix=daylight_matrix, facade_matrix=facade_matrix)
+        fph_cls.is_four_phase_calc = True
+        return fph_cls
+
+    def validate(self):
+        Command.validate(self)
+
+        calc_inputs_dict = {'is_direct_sun_calc': ('sun_coef_matrix', 'sun_vector'),
+                            'is_daylight_coef_calc': ('day_coef_matrix', 'sky_vector'),
+                            'is_three_phase_calc': ('view_matrix', 't_matrix',
+                                                    'daylight_matrix', 'sky_vector'),
+                            'is_four_phase_calc': (
+                            'view_matrix', 't_matrix', 'facade_matrix',
+                            'daylight_matrix', 'sky_vector')}
+
+        # Perform two checks:
+        #   Check1: Ensure that exactly one of the calc types are set to be run.
+        #   Check2: Ensure that all the inputs are available for the selected calc type.
+        # What can still go wrong:
+        #   If inputs that are not required for a particular calc type are set,
+        #   they will be ignored.
+
+        # Check 1
+        calc_list = [(key, getattr(self, key)) for key in calc_inputs_dict.keys()]
+        calc_set_true = [key for key, val in calc_list if val]
```

Comment:
This is why I think you should manage this under the hood. Maybe a single parameter that keeps track of the type.

`self._study_type = 'direct_sun'`

That should make the code cleaner and minimize the number of checks. You can use a check to make sure the user doesn't change the value once the class is initiated. The default can be set to the most common use case.

I know I'm leaving too many comments here. Let me know if you want me to give it a try!

Reply:
Okay. Regarding the default case, I'd rather not have a default case because dctimestep does get used a whole lot for a variety of purposes.